src/components/AvatarRow/AvatarRow.jsx

```diff
@@ -0,0 +1,168 @@
+    avatarProps.size || 'md'
+  ]
+  const avatarRowRef = useRef(null)
+  const [numberOfItemsOnDisplay, setNumberOfItemsOnDisplay] = useState(
+    avatars.length
+  )
+
+  const throttledOnResize = !ieCompatible
+    ? throttle(onResize, throttleWait)
+    : noop
+  const throttledHandleWindowResize = ieCompatible
+    ? throttle(handleWindowResize, throttleWait)
+    : noop
+
+  useEffect(() => {
+    if (!adaptable || ieCompatible || avatarRowRef.current == null) return
+
+    const avatarRowEl = avatarRowRef.current
+    const resizeObserver = setupObserver(
+      throttleOnResize ? throttledOnResize : onResize
+    )
+
+    resizeObserver.observe(avatarRowEl)
+
+    return () => {
+      if (!adaptable || ieCompatible || avatarRowEl == null) return
+
+      resizeObserver.unobserve(avatarRowEl)
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [])
+
+  useEffect(() => {
+    if (ieCompatible && adaptable) {
+      window.addEventListener(
+        'resize',
+        throttleOnResize ? throttledHandleWindowResize : handleWindowResize
+      )
+
+      const measures = avatarRowRef.current.getBoundingClientRect()
+      onResize(measures)
+    }
+
+    return () => {
+      ieCompatible &&
+        adaptable &&
+        window.addEventListener(
+          'resize',
+          throttleOnResize ? throttledHandleWindowResize : handleWindowResize
+        )
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [])
+
+  function handleWindowResize() {
+    if (avatarRowRef.current != null) {
+      const measures = avatarRowRef.current.getBoundingClientRect()
+
+      onResize(measures)
+    }
+  }
+
+  function onResize({ width: containerWidth }) {
+    /** Only act if we have more than 1 avatar */
+    if (!containerWidth || avatars.length <= 1) {
+      return
+    }
+
+    const spaceForAllAvatars =
+      avatarSize * avatars.length + (avatars.length - 1) * gap
+
+    if (containerWidth >= spaceForAllAvatars) {
+      setNumberOfItemsOnDisplay(avatars.length)
+    } else {
+      const numberOfGaps = numberOfItemsOnDisplay - 1
+      const itemsThatFit = Math.floor(
+        (containerWidth - numberOfGaps * gap) / avatarSize
+      )
+
+      setNumberOfItemsOnDisplay(itemsThatFit > 0 ? itemsThatFit : 1)
+    }
+  }
+
+  const { shownAvatars, hiddenAvatars } = splitAvatarsArray(
+    avatars,
+    numberOfItemsOnDisplay
+  )
+
+  return (
+    <AvatarRowUI
+      className={classNames('AvatarRow', className)}
+      data-cy={dataCy}
+      ref={avatarRowRef}
+      $gap={gap}
+      $minWidth={avatarSize}
+    >
+      {shownAvatars.map(props => (
+        <Avatar key={props.id || props.name} {...props} {...avatarProps} />
+      ))}
+      {hiddenAvatars.length > 0 && (
+        <Tooltip
+          title={hiddenAvatars.map(props => props.name).join('\n')}
+          withTriggerWrapper={false}
+          placement="bottom"
+          {...extraTooltipProps}
```

Comment:
And just for clarification...you mentioned something similar in the the Storybook file, but with `extraTooltipProps` we could provide a custom `title` value? Say if we wanted to say that certain users are typing.

Reply:
Yes, anything on `extraTooltipProps` overrides what is here